### PR TITLE
[IMP] web: evaluate string domain function handler

### PR DESCRIPTION
**Pull request by yosuanicolaus**
When getting a domain from activeFields in Owl, sometimes the value passed is not a list or function.

When the field mentioned have a `check_company=True` in the initialization, the domain passed will be in a form of string instead of a list.

This commit aims to publicize the domain handler in Field so that it can be used anywhere by calling the function, as getting field's domain is getting increasingly common

related enterprise-PR: https://github.com/odoo/enterprise/pull/49456

task-id: 3549961

### CODE REVIEW

**Comment by william-andre** [comment]
@FrancoisGe Wouldn't it be possible to use the `Field` component instead?
I played around with it a little but couldn't manage to do it

**Comment by FrancoisGe** [comment]
> @FrancoisGe Wouldn't it be possible to use the `Field` component instead? I played around with it a little but couldn't manage to do it

Hello
I don't think it's a good idea to modify Many2oneField. 
There are 2 solutions:
- Either use the Field component, in general we use the Field component only in views because it was built to convert a <field> from an arch into an OWL component. It manages the passage of attributes into props.
- Either pass a domain function to the <Many2oneField> which generates the domain. It might be worth creating a utils that we use in Field and in your case.

**Comment by william-andre** [comment]
> Either use the Field component

Just told you I didn't manage to do that :sweat_smile: 
The field would stay readonly, no matter what I tried.

**Comment by yosuanicolaus** [comment]
> * Either pass a domain function to the  which generates the domain. It might be worth creating a utils that we use in Field and in your case.

@FrancoisGe do you know an example for the domain function? (for reference)

**Comment by FrancoisGe** [comment]
> > * Either pass a domain function to the  which generates the domain. It might be worth creating a utils that we use in Field and in your case.
> 
> @FrancoisGe do you know an example for the domain function? (for reference)

You can do
```xml  
domain.bind="getMyDomain"
```
And you define getMyDomain in your js component :)

**Comment by william-andre** [comment]
@FrancoisGe I didn't expect we would need to do anything custom to get the default domain declared on the field :/

**Comment by yosuanicolaus** [comment]
well, I tried, and I think this is not the right way to go :/
it feels wrong to do so much effort at custom function in the custom JS component when I just want to get the original python field's domain...
There will be a lot of other fields that want to get the original field's domain in the future, so I still think we should improve the domain parser instead.

**Comment by FrancoisGe** [comment]
> @FrancoisGe I didn't expect we would need to do anything custom to get the default domain declared on the field :/

Did you try with <Field> ?

**Comment by yosuanicolaus** [comment]
@FrancoisGe yes :slightly_smiling_face: I was talking about the domain function. I tried using Field too but same like William, I can't make it not readonly

**Comment by FrancoisGe** [comment]
> @FrancoisGe yes :slightly_smiling_face: I was talking about the domain function. I tried using Field too but same like William, I can't make it not readonly

If you want, i Can check with you next week ?  Ping me on discord next week :)

File: addons/account/models/product.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 from odoo.osv import expression
 from odoo.tools import format_amount
 
-ACCOUNT_DOMAIN = "['&', '&', ('deprecated', '=', False), ('account_type', 'not in', ('asset_receivable','liability_payable','asset_cash','liability_credit_card','off_balance')), ('company_id', '=', current_company_id)]"
+ACCOUNT_DOMAIN = "['&', '&', ('deprecated', '=', False), ('account_type', 'not in', ('asset_receivable','liability_payable','asset_cash','liability_credit_card','off_balance')), ('company_id', 'parent_of', current_company_id)]"
```
Comment (william-andre) [inline addons/account/models/product.py:8]:
```suggestion
ACCOUNT_DOMAIN = "['&', '&', ('deprecated', '=', False), ('account_type', 'not in', ('asset_receivable','liability_payable','asset_cash','liability_credit_card','off_balance'))]"
```

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Comment by william-andre** [comment]
@robodoo r+

**Comment by william-andre** [comment]
@robodoo rebase-ff

**Pull request closed without merging**